homeassistant/components/yolink/cover.py

```diff
@@ -54,33 +55,21 @@ def __init__(
     @callback
     def update_entity_state(self, state: dict[str, Any]) -> None:
         """Update HA Entity State."""
-        self._attr_is_closed = state.get("state") == "closed"
+        if (state_val := state.get("state")) is None:
+            return
+        self._attr_is_closed = state_val == "closed"
         self.async_write_ha_state()
 
-    async def toggle_garage_state(self, state: str) -> None:
+    async def toggle_garage_state(self) -> None:
         """Toggle Garage door state."""
-        # make sure current state is correct
-        await self.coordinator.async_refresh()
-        if state == "open" and self.is_closed is False:
-            return
-        if state == "close" and self.is_closed is True:
-            return
-        # get paired controller
-        door_controller_coordinator = self.hass.data[DOMAIN][
-            self.config_entry.entry_id
-        ][ATTR_COORDINATORS].get(self.coordinator.device.parent_id)
-        if door_controller_coordinator is None:
-            raise ValueError(
-                "This device has not been paired with a garage door controller"
-            )
-        # call controller api open/close garage door
-        await door_controller_coordinator.device.call_device_http_api("toggle", None)
-        await self.coordinator.async_refresh()
+        # garage door state will not be change by device call
+        # it's depends on paired device state, such as door sensor or contact sensor
+        await self.call_device(ClientRequest("toggle", {}))
```

Comment:
Is there no specific request builder for cover and the other changed platforms similar to climate? Are you going to update them later?

Reply:
This PR change too much code, I want to update the others later.